drivers/overlay/overlay.go

```diff
@@ -2162,8 +2162,11 @@ func supportsDataOnlyLayersCached(home, runhome string) (bool, error) {
 // ApplyDiffWithDiffer applies the changes in the new layer using the specified function
 func (d *Driver) ApplyDiffWithDiffer(id, parent string, options *graphdriver.ApplyDiffWithDifferOpts, differ graphdriver.Differ) (output graphdriver.DriverWithDifferOutput, errRet error) {
 	var idMappings *idtools.IDMappings
+	forceMask := d.options.forceMask
+
 	if options != nil {
 		idMappings = options.Mappings
+		forceMask = options.ForceMask
```

Comment:
Should this be conditioned on `options.ForceMask != nil`?

(One point is that the semantics is not documented. Another is that this field is never set when called by c/image, anyway. And yet again, if it is not used, why is it a field of `ApplyDiffWithDifferOpts` at all…)

---

AFAICS `ApplyDiffOpts.ForceMask` is never set when creating non-staged/chunked layers, so maybe we can just remove the filed and simplify?

Reply:
(Running on macOS, apparently I can’t trust the IDE:) `ApplyDiff` does
```go
	if d.options.forceMask != nil {
			options.ForceMask = d.options.forceMask
		}
```
the opposite of this logic… but, also, the field is not used on that path at all, either.

Removing `ApplyDiffOpts.ForceMask` looks increasingly attractive to me.

Reply:
thanks for spotting it.  I'd say the options should have a higher precedence, but I've adapted it to follow what `ApplyDiff` does.

Maybe something more to simplify along the lines of https://github.com/containers/storage/pull/1976